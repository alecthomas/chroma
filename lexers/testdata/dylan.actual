Module: mod

define constant $pi = 3.1415927d0;

define macro moo
  { moo(?:name, ?e:expression) } => { let ?name = ?e; }
end;

// abc
define method foo-bar (n :: <integer>, #rest a, #key k, #all-keys) => (m :: <float>, b :: <boolean>)
  let x = n + 2.0 + #b10 + #o2 + #x02;
  format-out("dog // cat\n");
  local method pi () $pi end;
  pi()
end method foo-bar;

/*
 cow
 */
foo-bar(2, k: k, a-b: c);
