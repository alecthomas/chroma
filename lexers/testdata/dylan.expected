[
  {"type":"NameAttribute","value":"Module:"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralString","value":"mod"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Name","value":"define"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"constant"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameConstant","value":"$pi"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumberFloat","value":"3.1415927d0"},
  {"type":"Punctuation","value":";"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Name","value":"define"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"macro"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"moo"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"Punctuation","value":"{"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"moo"},
  {"type":"Punctuation","value":"("},
  {"type":"Operator","value":"?:"},
  {"type":"NameVariable","value":"name"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"?"},
  {"type":"NameVariable","value":"e"},
  {"type":"Operator","value":":"},
  {"type":"NameBuiltin","value":"expression"},
  {"type":"Punctuation","value":")"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"}"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"=\u003e"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltin","value":"let"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"?name"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"?"},
  {"type":"NameVariable","value":"e"},
  {"type":"Punctuation","value":";"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"}"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Name","value":"end"},
  {"type":"Punctuation","value":";"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"CommentSingle","value":"// abc\n"},
  {"type":"Name","value":"define"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltin","value":"method"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"foo-bar"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"("},
  {"type":"Name","value":"n"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"::"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameClass","value":"\u003cinteger\u003e"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"#rest"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"a"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"#key"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"k"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"#all-keys"},
  {"type":"Punctuation","value":")"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"=\u003e"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"("},
  {"type":"Name","value":"m"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"::"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameClass","value":"\u003cfloat\u003e"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"b"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"::"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameClass","value":"\u003cboolean\u003e"},
  {"type":"Punctuation","value":")"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"NameBuiltin","value":"let"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"x"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"n"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"+"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumberFloat","value":"2.0"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"+"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumberBin","value":"#b10"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"+"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumberOct","value":"#o2"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"+"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumberHex","value":"#x02"},
  {"type":"Punctuation","value":";"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"Name","value":"format-out"},
  {"type":"Punctuation","value":"("},
  {"type":"LiteralString","value":"\"dog // cat"},
  {"type":"LiteralStringEscape","value":"\\n"},
  {"type":"LiteralString","value":"\""},
  {"type":"Punctuation","value":");"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"Name","value":"local"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltin","value":"method"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"pi"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"()"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameConstant","value":"$pi"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"end"},
  {"type":"Punctuation","value":";"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"Name","value":"pi"},
  {"type":"Punctuation","value":"()"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Name","value":"end"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltin","value":"method"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"foo-bar"},
  {"type":"Punctuation","value":";"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"CommentMultiline","value":"/*\n cow\n */"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Name","value":"foo-bar"},
  {"type":"Punctuation","value":"("},
  {"type":"LiteralNumberInteger","value":"2"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"KeywordConstant","value":"k:"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"k"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"KeywordConstant","value":"a-b:"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"c"},
  {"type":"Punctuation","value":");"},
  {"type":"TextWhitespace","value":"\n"}
]
